internal/maintenance/queue_cleaner.go

```diff
@@ -0,0 +1,162 @@
+package maintenance
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log/slog"
+	"strings"
+	"time"
+
+	"github.com/riverqueue/river/internal/baseservice"
+	"github.com/riverqueue/river/internal/maintenance/startstop"
+	"github.com/riverqueue/river/internal/rivercommon"
+	"github.com/riverqueue/river/internal/util/timeutil"
+	"github.com/riverqueue/river/internal/util/valutil"
+	"github.com/riverqueue/river/riverdriver"
+)
+
+const (
+	QueueCleanerIntervalDefault = time.Hour
+	QueueRetentionPeriodDefault = 24 * time.Hour
+)
+
+// Test-only properties.
+type QueueCleanerTestSignals struct {
+	DeletedBatch rivercommon.TestSignal[struct{}] // notifies when runOnce finishes a pass
+}
+
+func (ts *QueueCleanerTestSignals) Init() {
+	ts.DeletedBatch.Init()
+}
+
+type QueueCleanerConfig struct {
+	// Interval is the amount of time to wait between runs of the cleaner.
+	Interval time.Duration
+	// QueueRetentionPeriod is the amount of time to keep queues around before
+	// they're removed.
+	QueueRetentionPeriod time.Duration
```

Comment:
Maybe drop this one down to just `RetentionPeriod`. The additional "Queue" stutters with what's implied by the struct's name.